addons/base_import/models/base_import.py

```diff
@@ -908,24 +1202,27 @@ def do(self, fields, columns, options, dryrun=False):
         self._cr.execute('SAVEPOINT import')
 
         try:
-            data, import_fields = self._convert_import_data(fields, options)
+            input_file_data, import_fields = self._convert_import_data(fields, options)
             # Parse date and float field
-            data = self._parse_import_data(data, import_fields, options)
-        except ValueError as error:
-            return {
-                'messages': [{
-                    'type': 'error',
-                    'message': str(error),
-                    'record': False,
-                }]
-            }
+            input_file_data = self._parse_import_data(input_file_data, import_fields, options)
+        except ImportValidationError as error:
+            return error.error
+
+        _logger.info('importing %d rows...', len(input_file_data))
+
+        import_fields, merged_data = self._handle_multi_mapping(import_fields, input_file_data)
 
-        _logger.info('importing %d rows...', len(data))
+        if options.get('fallback_values'):
+            merged_data = self._handle_fallback_values(import_fields, merged_data, options['fallback_values'])
 
         name_create_enabled_fields = options.pop('name_create_enabled_fields', {})
         import_limit = options.pop('limit', None)
-        model = self.env[self.res_model].with_context(import_file=True, name_create_enabled_fields=name_create_enabled_fields, _import_limit=import_limit)
-        import_result = model.load(import_fields, data)
+        model = self.env[self.res_model].with_context(
+            import_file=True,
+            name_create_enabled_fields=name_create_enabled_fields,
+            import_skip_fields=options.get('import_skip_fields'),
+            _import_limit=import_limit)
+        import_result = model.load(import_fields, merged_data)
```

Comment:
Since you've got your fingers in that anyway it might be a good idea to fix task 2274253 while at it?

Basically move the "importing..." bit to the "load" method and log *the model*, the number of *records* (instead of rows), and the user. Because currently it's possible to see that an import is happening, but not an import of what or by whom, which can make tracking large imports harder.

Reply:
I propose to let that improvement in that task who has been canceled.